hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/pages/buckets/buckets.tsx

```diff
@@ -0,0 +1,563 @@
+    render: (text: string, record: Bucket) => (
+      <QuotaBar
+        quota={record.quotaInNamespace}
+        used={record.usedNamespace}
+        quotaType='namespace'
+      />
+    )
+  },
+  {
+    title: 'Source Volume',
+    dataIndex: 'sourceVolume',
+    key: 'sourceVolume',
+    render: (sourceVolume: string) => {
+      return sourceVolume ? sourceVolume : 'NA';
+    }
+  },
+  {
+    title: 'Source Bucket',
+    dataIndex: 'sourceBucket',
+    key: 'sourceBucket',
+    render: (sourceBucket: string) => {
+      return sourceBucket ? sourceBucket : 'NA';
+    }
+  }
+];
+
+const defaultColumns = COLUMNS.map(column => ({
+  label: column.title as string,
+  value: column.key as string
+}));
+
+function getVolumeBucketMap(data: Bucket[]) {
+  const volumeBucketMap = data.reduce((
+    map: Map<string, Set<Bucket>>,
+    currentBucket
+  ) => {
+    const volume = currentBucket.volumeName;
+    if (map.has(volume)) {
+      const buckets = Array.from(map.get(volume)!);
+      map.set(volume, new Set([...buckets, currentBucket]));
+    } else {
+      map.set(volume, new Set<Bucket>().add(currentBucket));
+    }
+    return map;
+  }, new Map<string, Set<Bucket>>());
+  return volumeBucketMap;
+}
+
+function getFilteredBuckets(
+  selectedVolumes: Option[],
+  bucketsMap: Map<string, Set<Bucket>>
+) {
+  let selectedBuckets: Bucket[] = [];
+  selectedVolumes.forEach(selectedVolume => {
+    if (bucketsMap.has(selectedVolume.value)
+      && bucketsMap.get(selectedVolume.value)) {
+      selectedBuckets = [
+        ...selectedBuckets,
+        ...Array.from(bucketsMap.get(selectedVolume.value)!)
+      ];
+    }
+  });
+
+  return selectedBuckets;
+}
+
+const Buckets: React.FC<{}> = () => {
+
+  let cancelSignal: AbortController;
+
+  const [state, setState] = useState<BucketsState>({
+    totalCount: 0,
+    lastUpdated: 0,
+    columnOptions: defaultColumns,
+    volumeBucketMap: new Map<string, Set<Bucket>>(),
+    bucketsUnderVolume: [],
+    volumeOptions: [],
+  });
+  const [loading, setLoading] = useState<boolean>(false);
+  const [selectedColumns, setSelectedColumns] = useState<Option[]>(defaultColumns);
+  const [selectedVolumes, setSelectedVolumes] = useState<Option[]>([]);
+  const [selectedLimit, setSelectedLimit] = useState<Option>(LIMIT_OPTIONS[0]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [showPanel, setShowPanel] = useState<boolean>(false);
+  const [searchColumn, setSearchColumn] = useState<'name' | 'volumeName'>('name');
+  const [currentRow, setCurrentRow] = useState<Bucket | Record<string, never>>({})
+
+  const debouncedSearch = useDebounce(searchTerm, 300);
+  const { search } = useLocation();
+
+  const paginationConfig: TablePaginationConfig = {
+    showTotal: (total: number, range) => `${range[0]}-${range[1]} of ${total} buckets`,
+    showSizeChanger: true
+  };
+
+  function getVolumeSearchParam() {
+    return new URLSearchParams(search).get('volume');
```

Comment:
Yes it will be null as it is the search param which comes only when we click on `Show Buckets` in the Volumes page.
If we directly access the Buckets page, this param will not be present and hence return null.
We are handling this already in the code:
https://github.com/apache/ozone/blob/c29db4c59a7a6d622c49b18ae3eee65b4e481a10/hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/pages/buckets/buckets.tsx#L452